src/simulators/statevector/statevector_state_chunk.hpp

```diff
@@ -480,6 +472,35 @@ auto State<statevec_t>::move_to_vector()
   }
 }
 
+template <class statevec_t>
+auto State<statevec_t>::copy_to_vector()
+{
+  if(BaseState::num_global_chunks_ == 1){
+    return BaseState::qregs_[0].copy_to_vector();
+  }
+  else{
+    int_t iChunk;
+    auto state = BaseState::qregs_[0].copy_to_vector();
+
+    //TO DO check memory availability
+    state.resize(BaseState::num_local_chunks_ << BaseState::chunk_bits_);
+
+#pragma omp parallel for if(BaseState::chunk_omp_parallel_) private(iChunk)
+    for(iChunk=1;iChunk<BaseState::num_local_chunks_;iChunk++){
+      auto tmp = BaseState::qregs_[iChunk].copy_to_vector();
+      uint_t j,offset = iChunk << BaseState::chunk_bits_;
+      for(j=0;j<tmp.size();j++){
+        state[offset + j] = tmp[j];
```

Comment:
Same doubt as in `densitymatrix_chunk`? cannot we directly copy from `BaseState::qregs_[iChunk]` to `state`, avoiding the copy to `tmp`.

Reply:
this is same reason as density matrix